lib/blocktree/blocktree_test.go

```diff
@@ -371,7 +371,141 @@ func TestBlockTree_IsDecendantOf(t *testing.T) {
 	require.False(t, isDescendant)
 }
 
-func TestBlockTree_HighestCommonAncestor(t *testing.T) {
+func Test_Node_LowestCommonAncestorNew(t *testing.T) {
+	t.Parallel()
+	root := &node{
+		hash:   common.Hash{0},
+		number: 0,
+	}
+
+	children := []*node{
+		{
+			hash:      common.Hash{1},
+			parent:    root,
+			isPrimary: true,
+			number:    1,
+		},
+		{
+			hash:      common.Hash{2},
+			parent:    root,
+			isPrimary: false,
+			number:    1,
+		},
+	}
+
+	childrenChildren := []*node{
+		{
+			hash:      common.Hash{3},
+			parent:    children[0],
+			isPrimary: true,
+			number:    2,
+		},
+		{
+			hash:      common.Hash{4},
+			parent:    children[1],
+			isPrimary: false,
+			number:    2,
+		},
+	}
+	finalChild := []*node{
+		{
+			hash:      common.Hash{5},
+			parent:    childrenChildren[1],
+			isPrimary: true,
+			number:    3,
+		},
+		{
+			hash:      common.Hash{6},
+			parent:    childrenChildren[1],
+			isPrimary: true,
+			number:    21,
+		},
+	}
+
+	type args struct {
+		nodeA *node
+		nodeB *node
+	}
+	tests := []struct {
+		name      string
+		args      args
+		expErr    error
+		expErrMsg string
+		expRes    Hash
+	}{
+		{
+			name: "child and root",
+			args: args{
+				nodeA: children[1],
+				nodeB: root,
+			},
+			expRes: root.hash,
+		},
+		{
+			name: "same node",
+			args: args{
+				nodeA: children[1],
+				nodeB: children[1],
+			},
+			expRes: children[1].hash,
+		},
+		{
+			name: "siblings",
+			args: args{
+				nodeA: children[0],
+				nodeB: children[1],
+			},
+			expRes: root.hash,
+		},
+		{
+			name: "child and its child",
+			args: args{
+				nodeA: children[0],
+				nodeB: childrenChildren[0],
+			},
+			expRes: children[0].hash,
+		},
+		{
+			name: "root and grandchild",
+			args: args{
+				nodeA: root,
+				nodeB: childrenChildren[0],
+			},
+			expRes: root.hash,
+		},
+		{
+			name: "grandchild and its siblings child",
+			args: args{
+				nodeA: finalChild[0],
+				nodeB: childrenChildren[0],
+			},
+			expRes: root.hash,
+		},
+		{
+			name: "out of bounds error",
+			args: args{
+				nodeA: finalChild[1],
+				nodeB: childrenChildren[0],
+			},
+			expErr:    errorAncestorOutOfBoundsCheck,
+			expErrMsg: "out of bounds ancestor check: for block number 21",
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			ancestor, err := lowestCommonAncestor(tt.args.nodeA, tt.args.nodeB)
+			assert.ErrorIs(t, err, tt.expErr)
+			if tt.expErr != nil {
+				assert.EqualError(t, err, tt.expErrMsg)
```

Comment:
This seems like overkill, since we already do the `assert.ErrorIs` check.  I don't think we need to string comparisons on error messages.